banyand/stream/query.go

```diff
@@ -41,6 +43,15 @@ type queryOptions struct {
 	maxTimestamp int64
 }
 
+func mustEncodeTagValue(name string, tagType databasev1.TagType, tagValue *modelv1.TagValue, num int) [][]byte {
+	values := make([][]byte, num)
+	nv := encodeTagValue(name, tagType, tagValue)
+	for i := 0; i < num; i++ {
+		values[i] = nv.marshal()
```

Comment:
Moving the marshaling process outside the loop will help improve performance and efficiency.